drivers/wireless/lpwan/rn2483/rn2483.c

```diff
@@ -0,0 +1,862 @@
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set sf (Spread Factor)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_SF, &priv->config.sf);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set sync (Synchronization word)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_SYNC, &priv->config.sync);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set mod (Modulation: FSK or LoRa)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_MOD, &priv->config.mod);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  return F_OK;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: rn2483_register
+ *
+ * Description:
+ *   Register the RN2483 LoRa driver.
+ *
+ ****************************************************************************/
+
+int rn2483_register(FAR const char *devpath, FAR const char *uartpath)
+{
+  syslog(LOG_INFO, "Enter Register\n\n");
+
+  FAR struct rn2483_dev_s *priv = NULL;
+  int err = 0;
+
+  DEBUGASSERT(uartpath != NULL);
+
+  /* Initialize the device structure */
+
+  priv = kmm_zalloc(sizeof(struct rn2483_dev_s));
+  if (priv == NULL)
+  {
+    snerr("ERROR: Failed to allocate instance.\n");
+    return -ENOMEM;
+  }
+
+  /* Initialize mutex */
+
+  err = nxmutex_init(&priv->devlock);
+  if (err < 0)
+  {
+    // TODO: what logging macro needs to be used
+    // snerr("ERROR: Failed to register SHT4X driver: %d\n", err);
+    kmm_free(priv);
+    return err;
+  }
+
+  /* Get access to underlying UART interface */
+
+  err = file_open(&priv->uart, uartpath, O_RDWR | O_CLOEXEC);
+  if (err < 0)
+  {
+    // TODO log error
+    nxmutex_destroy(&priv->devlock);
+    kmm_free(priv);
+    return err;
+  }
+
+  /* Set configuration options. */
+
+#if CONFIG_LPWAN_RN2483_CR == 5
+  priv->config.cr = RN2483_CR_4_5;
+#elif CONFIG_LPWAN_RN2483_CR == 6
+  priv->config.cr = RN2483_CR_4_6;
+#elif CONFIG_LPWAN_RN2483_CR == 7
+  priv->config.cr = RN2483_CR_4_7;
+#elif CONFIG_LPWAN_RN2483_CR == 8
+  priv->config.cr = RN2483_CR_4_8;
+#endif
+
+#if CONFIG_LPWAN_RN2483_LORA
+  priv->config.mod = RN2483_MOD_LORA;
+#else
+  priv->config.mod = RN2483_MOD_FSK;
+#endif
+
+  priv->config.freq = CONFIG_LPWAN_RN2483_FREQ;
+
+  // Error check of frequency value (Not sure if this is what you want Matteo)
+  assert((priv->config.freq >= 433000000 && priv->config.freq <= 434800000) || (priv->config.freq >= 863000000 && priv->config.freq <= 870000000));
+
+  priv->config.bw = CONFIG_LPWAN_RN2483_BW;
+  priv->config.prlen = CONFIG_LPWAN_RN2483_PRLEN;
+  priv->config.iqi = CONFIG_LPWAN_RN2483_IQI;
+  priv->config.crc = CONFIG_LPWAN_RN2483_CRC;
+  priv->config.pwr = CONFIG_LPWAN_RN2483_POWER;
+  priv->config.sf = CONFIG_LPWAN_RN2483_SPREAD;
+  priv->config.sync = CONFIG_LPWAN_RN2483_SYNC;
+
+  // Dummy read to get rid of version string
```

Comment:
Like mentioned higher in the file, the reading logic should be extracted into another function so it can be re-used in multiple places. We'll also need to brainstorm a way to detect whether or not the version string was sent on power up or if we booted to late to receive it, because if we read when the buffer is empty we'll hang.